File: frontend/src/lib/api/generative_ui.ts
import { request } from "./client";

export interface GenerativeInterface {
    id: string;
    title: string;
    description: string | null;
    prompt: string;
    content_html: string;
    is_pinned: boolean;
    created_at: string;
    updated_at: string;
}

export const generativeUI = {
    generate: (prompt: string) =>
        request<GenerativeInterface>("/api/v1/generative-ui/generate", {
            method: "POST",
            body: JSON.stringify({ prompt }),
        }),

    history: () =>
        request<GenerativeInterface[]>("/api/v1/generative-ui/history"),

    get: (id: string) =>
        request<GenerativeInterface>(`/api/v1/generative-ui/${id}`),

    update: (id: string, data: { title?: string; description?: string }) =>
        request<GenerativeInterface>(`/api/v1/generative-ui/${id}`, {
            method: "PATCH",
            body: JSON.stringify(data),
        }),

    delete: (id: string) =>
        request<void>(`/api/v1/generative-ui/${id}`, { method: "DELETE" }),
};

File: backend/app/api/v1/schemas/generative_ui.py
"""Schemas Pydantic para Generative UI."""

from pydantic import BaseModel, ConfigDict, Field


class GenerateRequest(BaseModel):
    prompt: str = Field(max_length=8000)
    title: str | None = None


class GeneratedUIOut(BaseModel):
    id: str
    title: str
    description: str | None
    prompt: str
    content_html: str
    is_pinned: bool
    created_at: str
    updated_at: str

    model_config = ConfigDict(from_attributes=True)


class UpdateUIRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    is_pinned: bool | None = None
